src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3748,7 +3770,16 @@ bool os::Linux::setup_large_page_type(size_t page_size) {
     UseHugeTLBFS = false;
   }
 
-  return UseSHM;
+  if (UseSHM) {
+    bool warn_on_failure = !FLAG_IS_DEFAULT(UseSHM);
+    if (shm_hugetlbfs_sanity_check(warn_on_failure, page_size)) {
+      return true;
+    }
+    UseSHM = false;
+  }
+
+  log_warning(pagesize)("UseLargePages disabled, no large pages configured and available on the system.");
```

Comment:
IIUC here we end up if UseLargePages=true (by default or not) and we were unable to get any of our APIs to work? Should this be an unconditional printout? At least make it only unconditional if UseLargePages==true is not default? I know its only a theoretical problem now since default is false.

Reply:
You are correct we only end up here if `UseLargePages` is true. Making it conditional like that would be consistent with the other warnings. So only warn if the user explicitly set `+UseLargePages`. So something like?
```suggestion
  if (!FLAG_IS_DEFAULT(UseLargePages)) {
    log_warning(pagesize)("UseLargePages disabled, no large pages configured and available on the system.");
  }
```

Reply:
Seems reasonable and in line with what we do otherwise.

As a side note, I'm not a big fan of unconditional error printouts for non-fatal errors. Can mess with parsers (see eg JDK-8260902). The argument "we trace only if flag is explicitly set" is valid oc, but even so customers tend to bury flags in scripts and then forget them. But I know this is common procedure so this would be a separate discussion.